pkg/providers/oidc/verifier_test.go

```diff
@@ -181,12 +181,82 @@ var _ = Describe("Verify", func() {
 		Expect(result.Issuer).To(Equal("https://foo"))
 		Expect(result.Audience).To(Equal([]string{"1226737"}))
 	})
+
+	It("Succeeds with default email_verified behavior", func() {
+		verified := true
+		_, err := verify(ctx, IDTokenVerificationOptions{
+			AudienceClaims: []string{"aud"},
+			ClientID:       "1226737",
+			ExtraAudiences: []string{},
+		}, payload{
+			Iss:      "https://foo",
+			Aud:      "1226737",
+			Email:    "test@example.com",
+			Verified: &verified,
+		})
+
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("Succeeds if unverified email is allowed", func() {
+		verified := false
+		_, err := verify(ctx, IDTokenVerificationOptions{
+			AudienceClaims:       []string{"aud"},
+			ClientID:             "1226737",
+			ExtraAudiences:       []string{},
+			AllowUnverifiedEmail: true,
+		}, payload{
+			Iss:      "https://foo",
+			Aud:      "1226737",
+			Email:    "test@example.com",
+			Verified: &verified,
+		})
+
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("Fails with with default email_verified behavior", func() {
```

Comment:
```suggestion
	It("Fails with unverified email if AllowUnverifiedEmail is NOT set", func() {
```

In theory this test case isn't necessary because booleans are implicitly false by default nevertheless it doesn't hurt to keep it.